yt_dlp/extractor/instagram.py

```diff
@@ -647,22 +648,22 @@ class InstagramStoryIE(InstagramBaseIE):
 
     def _real_extract(self, url):
         username, story_id = self._match_valid_url(url).groups()
-
-        story_info, wh = self._download_webpage_handle(url, story_id)
-        if 'www.instagram.com/accounts/login' in wh.geturl():
-            self.raise_login_required('You need to log in to access this content')
+        story_info = self._download_webpage(url, story_id)
         user_info = self._search_json(r'"user":', story_info, 'user info', story_id, fatal=False)
         if not user_info:
             self.raise_login_required('This content is unreachable')
         user_id = user_info.get('id')
 
         story_info_url = user_id if username != 'highlights' else f'highlight:{story_id}'
-        videos = self._download_json(f'https://i.instagram.com/api/v1/feed/reels_media/?reel_ids={story_info_url}', story_id, headers={
+        video_data = self._download_json(f'https://i.instagram.com/api/v1/feed/reels_media/?reel_ids={story_info_url}', story_id, headers={
             'X-IG-App-ID': 936619743392459,
             'X-ASBD-ID': 198387,
             'X-IG-WWW-Claim': 0,
-        })['reels']
+        }, errnote=False, fatal=False)
+        if not video_data:
+            self.raise_login_required('You need to log in to access this content')
 
+        videos = video_data.get('reels')
```

Comment:
why .get?

Reply:
The contributing best practices mentions to use .get and isn't it better to use it rather than using ['reels'].